cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ListenersUtils.java

```diff
@@ -0,0 +1,489 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.strimzi.api.kafka.model.listener.KafkaListenerAuthenticationOAuth;
+import io.strimzi.api.kafka.model.listener.NodeAddressType;
+import io.strimzi.api.kafka.model.listener.v2.GenericKafkaListener;
+import io.strimzi.api.kafka.model.listener.v2.GenericKafkaListenerConfigurationBroker;
+import io.strimzi.api.kafka.model.listener.v2.KafkaListenerType;
+import io.strimzi.api.kafka.model.template.ExternalTrafficPolicy;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Util methods for working with Kafka listeners
+ */
+public class ListenersUtils {
+    /*test*/ static final String BACKWARDS_COMPATIBLE_PLAIN_PORT_NAME = "tcp-clients";
+    /*test*/ static final String BACKWARDS_COMPATIBLE_TLS_PORT_NAME = "tcp-clientstls";
+    /*test*/ static final String BACKWARDS_COMPATIBLE_EXTERNAL_PORT_NAME = "tcp-external";
+
+    /**
+     * Finds out if any of the listeners has OAuth authentication enabled
+     *
+     * @param listeners List of all listeners
+     * @return          List of used names
+     */
+    public static boolean hasListenerWithOAuth(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> listener.getAuth() != null && KafkaListenerAuthenticationOAuth.TYPE_OAUTH.equals(listener.getAuth().getType()))
+                .map(listener -> true)
+                .findFirst()
+                .orElse(false);
+    }
+
+    /**
+     * Returns list of all internal listeners
+     *
+     * @param listeners List of all listeners
+     * @return          List of internal listeners
+     */
+    public static List<GenericKafkaListener> internalListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.INTERNAL == listener.getType())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of all external listeners (i.e. not internal)
+     *
+     * @param listeners List of all listeners
+     * @return          List of external listeners
+     */
+    public static List<GenericKafkaListener> externalListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.INTERNAL != listener.getType())
+                .collect(Collectors.toList());
```

Comment:
Factor out a private method which takes the desired `KafkaListenerType` as a param then each of these methods becomes a trivial `return privateMethod(KafkaListenerType.XXX);`